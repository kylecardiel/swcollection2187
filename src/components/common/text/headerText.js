import { makeStyles } from '@material-ui/core/styles';
import React from 'react';

const useStyles = makeStyles(theme => ({
    textSizeChange: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        fontFamily: 'Raleway, sans-serif',
        [theme.breakpoints.down('md')]: {
            fontSize: '18px',
        },
        [theme.breakpoints.up('md')]: {
            fontSize: '36px',
        },
        fontWeight: '800',
        textTransform: 'uppercase',
        color: props => props.textColor
    },
    textStatic: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        fontFamily: 'Raleway, sans-serif',
        fontSize: '36px',
        fontWeight: '800',
        textTransform: 'uppercase',
        color: props => props.textColor
    },
}));

export const HeaderText = props => {

    const { textColor, staticTextSize } = props;
    const classes = useStyles({ textColor: textColor ? textColor : 'black'});

    return (
        <div data-testid='headerTextId' className={staticTextSize ? classes.textStatic : classes.textSizeChange}>
            {props.text}
        </div>
    );
}
