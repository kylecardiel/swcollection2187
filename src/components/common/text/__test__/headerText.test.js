import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { HeaderText } from 'components/common/text/headerText';

describe('HeaderText', () => {

    const text = 'This is the Header';

    afterEach(cleanup);

    it('renders component', () => {
        const { asFragment } = render(<HeaderText text={text}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('inserts text into div', () => {
        const { getByTestId } = render(<HeaderText text={text}/>);
        expect(getByTestId('headerTextId')).toHaveTextContent(text);
    });

    describe('renders className', () => {

        it('adds correct className when staticTextSize is passed', () => {
            const { getByText } = render(<HeaderText text={text} textColor='red' staticTextSize/>);
            expect(getByText(text)).toHaveClass('makeStyles-textStatic-10 makeStyles-textStatic-12');
        });

        it('adds correct className when staticTextSize is not passed', () => {
            const { getByText } = render(<HeaderText text={text} textColor='red'/>);
            expect(getByText(text)).toHaveClass('makeStyles-textSizeChange-13 makeStyles-textSizeChange-15');
        });

    });

    it('styles text with correct color', () => {
        const { getByTestId } = render(<HeaderText text={text} textColor='red'/>);
        expect(getByTestId('headerTextId')).toHaveStyle('color: red')
    });

    it('styles accurately', () => {
        const { getByTestId } = render(<HeaderText text={text} textColor='green' staticTextSize/>);
        const testingComponent = getByTestId('headerTextId');
        expect(testingComponent).toHaveStyle('color: green');
        expect(testingComponent).toHaveStyle('display: flex');
        expect(testingComponent).toHaveStyle('flex-direction: column');
        expect(testingComponent).toHaveStyle('align-items: center');
        expect(testingComponent).toHaveStyle('font-family: Raleway, sans-serif');
        expect(testingComponent).toHaveStyle('font-size: 36px');
        expect(testingComponent).toHaveStyle('font-weight: 800');
        expect(testingComponent).toHaveStyle('text-transform: uppercase');
    });

});