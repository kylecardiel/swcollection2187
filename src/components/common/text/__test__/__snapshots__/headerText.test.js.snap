// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderText renders component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-textSizeChange-1 makeStyles-textSizeChange-3"
    data-testid="headerTextId"
  >
    This is the Header
  </div>
</DocumentFragment>
`;
